data/data/vsphere/main.tf

```diff
@@ -0,0 +1,77 @@
+provider "vsphere" {
+  user                 = var.vsphere_username
+  password             = var.vsphere_password
+  vsphere_server       = var.vsphere_url
+  allow_unverified_ssl = true
+}
+
+data "vsphere_datacenter" "datacenter" {
+  name = var.vsphere_datacenter
+}
+
+data "vsphere_compute_cluster" "cluster" {
+  name          = var.vsphere_cluster
+  datacenter_id = data.vsphere_datacenter.datacenter.id
+}
+
+data "vsphere_datastore" "datastore" {
+  name          = var.vsphere_datastore
+  datacenter_id = data.vsphere_datacenter.datacenter.id
+}
+
+data "vsphere_network" "network" {
+  name          = var.vsphere_network
+  datacenter_id = data.vsphere_datacenter.datacenter.id
+}
+
+data "vsphere_virtual_machine" "template" {
+  name          = var.vsphere_template
+  datacenter_id = data.vsphere_datacenter.datacenter.id
+}
+
+resource "vsphere_folder" "folder" {
+  path          = var.vsphere_folder
+  type          = "vm"
+  datacenter_id = data.vsphere_datacenter.datacenter.id
+}
+
+resource "vsphere_resource_pool" "resource_pool" {
+  name                    = var.vsphere_resource_pool
+  parent_resource_pool_id = data.vsphere_compute_cluster.cluster.resource_pool_id
+}
+
+module "bootstrap" {
+  source = "./bootstrap"
+
+  ignition      = var.ignition_bootstrap
+  resource_pool = vsphere_resource_pool.resource_pool.id
+  datastore     = data.vsphere_datastore.datastore.id
+  folder        = vsphere_folder.folder.id
+  network       = data.vsphere_network.network.id
+  datacenter    = data.vsphere_datacenter.datacenter.id
+  template      = data.vsphere_virtual_machine.template.id
+  guest_id      = data.vsphere_virtual_machine.template.guest_id
+}
+
+module "master" {
+  source = "./master"
+
+  // limitation of baremetal-runtimecfg.  The hostname must be master
+  name           = "master"
```

Comment:
would like to see this documented in the enhancement plus in the commit message.